In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

from sklearn import linear_model
from sklearn.linear_model import LinearRegression
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import PolynomialFeatures
from sklearn.naive_bayes import MultinomialNB
classifier = MultinomialNB()
from sklearn.cross_validation import train_test_split
from sklearn.feature_extraction.text import CountVectorizer

In [2]:
%matplotlib inline

In [8]:
spamdata = pd.read_csv('spambase.data', header=None)
spamdata.head()

0     1     2   3     4     5     6     7     8     9  ...    48     49  \
0  0.00  0.64  0.64   0  0.32  0.00  0.00  0.00  0.00  0.00 ...  0.00  0.000   
1  0.21  0.28  0.50   0  0.14  0.28  0.21  0.07  0.00  0.94 ...  0.00  0.132   
2  0.06  0.00  0.71   0  1.23  0.19  0.19  0.12  0.64  0.25 ...  0.01  0.143   
3  0.00  0.00  0.00   0  0.63  0.00  0.31  0.63  0.31  0.63 ...  0.00  0.137   
4  0.00  0.00  0.00   0  0.63  0.00  0.31  0.63  0.31  0.63 ...  0.00  0.135   

   50     51     52     53     54   55    56  57  
0   0  0.778  0.000  0.000  3.756   61   278   1  
1   0  0.372  0.180  0.048  5.114  101  1028   1  
2   0  0.276  0.184  0.010  9.821  485  2259   1  
3   0  0.137  0.000  0.000  3.537   40   191   1  
4   0  0.135  0.000  0.000  3.537   40   191   1  

[5 rows x 58 columns]

#### Subsample the data set so 60% is training data and 40% is test data. 

In [5]:
pop = spamdata.pop(57)
spam_train_X, spam_test_X, ham_train_y, ham_test_y = train_test_split(spamdata, pop, random_state=32, train_size=.6)

#### Write code to classify the data into spam and not-spam.

Training with training data

In [6]:
classifier.fit(spam_train_X, ham_train_y)
classifier.score(spam_train_X, ham_train_y)

0.79528985507246375

Testing with test data

In [7]:
classifier.fit(spam_test_X, ham_test_y)
classifier.score(spam_test_X, ham_test_y)

0.79033134166214014

### Conclusion:
The model performed well. The machine was able to learn on the training data (with a score of .795), which was 60% of the actual data, and so it performed well with the test data (score of .790), which was the remaining 40% of the actual data.